### DISCUSSION TITLE
How might I change the unit for a currency (bitcoin -> sats)?

### DISCUSSION THREAD

**Discussion by ghost** [firefly-iii/firefly-iii#6468] [General]
I have a few accounts where I track Bitcoin transactions. I am considering trying to switch the unit from Bitcoin to sats. In case you're not familiar, a sat is 1/100,000,000th of a bitcoin. So instead of inputting 0.0000025 I would input 250.

Because I have a lot of existing data, in order to do this, I assume I would have to muck with values in the database. Would it be enough to do

```sql
update transactions set amount = amount / 100000000;
```

The two potential problems with this I can think of are:

1. I would need to make sure I'm not introducing any floating point inaccuracy with that sort of division.
2. Are there other tables that would need to be updated?

**Comment by JC5** [comment]
That should do it for the most important part. Assuming you only manage BTC in your installation. 

Possibly, you need to adjust budgets, piggy banks, bills, etc.

The division should be no problem, but make a backup and try it is all I can suggest here.

**Comment by ghost** [comment]
Well I caught two obvious mistakes I was about to make. Query should be

```sql
update transactions set amount = amount * 100000000 where transaction_currency_id = 21;
```

The `where transaction_currency_id = 21` I figured out before running the query. Now what you said makes more sense to me ("Assuming you only manage BTC in your installation.").

The other mistake was using division instead of multiplication. Thankfully I used a transaction `begin; update ...` and inspected the results before committing; saved a lot of hassle. MySQL gave me a warning about truncating data when I tried the first query.

Looks like this is working, so thanks for the feedback.
